### ISSUE TITLE
Remove is_docker stuff from roles

### ISSUE THREAD

**Issue by rhysmeister** [ansible-collections/community.mongodb#217]
##### SUMMARY

Use ansible_facts.virtualization_type=='docker' as suggested by @cognifloyd 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
All roles

**Comment by adi-mohan** [comment]
Hi @rhysmeister ! May I work on this issue? Also, it would be great if you tag hacktoberfest label to this issue. It will help me in achieving hacktoberfest target. Thanks.

**Comment by rhysmeister** [comment]
@adi-mohan Hello, yes of course although there's no much info on here. Original discussion was here...

https://github.com/ansible-collections/community.mongodb/pull/216

**Comment by rhysmeister** [comment]
Here's all the files mentioning the is_docker.txt file...

```
(venv) Rhyss-MacBook-Pro:mongodb rhyscampbell$ grep -r "is_docker" roles/
roles//mongodb_linux/tasks/main.yml:    path: is_docker.txt
roles//mongodb_linux/molecule/default/tests/test_default.py:    is_docker = host.file("./is_docker.txt").exists
roles//mongodb_linux/molecule/default/tests/test_default.py:    if is_docker is False:
roles//mongodb_linux/molecule/default/prepare.yml:    - name: Create is_docker.txt file
roles//mongodb_linux/molecule/default/prepare.yml:      shell: touch is_docker.txt
roles//mongodb_linux/molecule/virtualbox/tests/test_default.py:    is_docker = host.file("./is_docker.txt").exists
roles//mongodb_linux/molecule/virtualbox/tests/test_default.py:    if is_docker is False:
roles//mongodb_mongod/tasks/main.yml:    path: is_docker.txt
roles//mongodb_config/tasks/main.yml:    path: is_docker.txt
roles//mongodb_config/molecule/default/prepare.yml:  - name: Create is_docker.txt file
roles//mongodb_config/molecule/default/prepare.yml:      path: is_docker.txt
```

So all those would need to be removed and replaced with Jacob's method. The python units tests would need a similar method of figuring out if it's currently in docker.

Cheers,

Rhys

**Comment by meysam81** [comment]
I was trying to figure out what should be done. Could you please guide me in the right direction? Perhaps with a little example would be great.

**Comment by cognifloyd** [comment]
In the prepare playbooks, remove the is_docker.txt file creation.
In the role tasks, remove the task that checks for is_docker.txt and modify vars to use ansible_facts to see if the host is in docker instead of relying on the is_docker.txt stat task.
In the molecule test_default.py you'll need to check the same ansible_facts var instead of looking for the is_docker.txt file.

**Comment by meysam81** [comment]
@cognifloyd 
Thanks for the tip.
I have done the first two steps (I guess), but for `test_default.py` I am not sure how to get an ansible_fact from Python script. Is it as simple as `os.getenv("ansible_facts.virtualization_type")`?

**Comment by rhysmeister** [comment]
See here....

https://testinfra.readthedocs.io/en/latest/modules.html#ansible

There's an example using Ansible Facts

**Comment by adi-mohan** [comment]
Should i comment out the is_docker task in main.yml or completely remove them? As commenting out will enable any future use. Also, prepare.yml won't have any tasks as such, if we remove is_docker file creation. So, should I remove the files entirely and also remove the calling task at tests/ansible-operator/molecule/kind/molecule.yml? Or is it a part of convention to have these files, whether-or-not tasks are there?

**Comment by adi-mohan** [comment]
Also, I am working on this issue, and will come up with the PR. As the issue was assigned to me, so I should be given chance to PR first. Thanks.

**Comment by rhysmeister** [comment]
> Should i comment out the is_docker task in main.yml or completely remove them? As commenting out will enable any future use. Also, prepare.yml won't have any tasks as such, if we remove is_docker file creation. So, should I remove the files entirely and also remove the calling task at tests/ansible-operator/molecule/kind/molecule.yml? Or is it a part of convention to have these files, whether-or-not tasks are there?

Take it out. New way is better.

**Comment by rhysmeister** [comment]
> Also, I am working on this issue, and will come up with the PR. As the issue was assigned to me, so I should be given chance to PR first. Thanks.

All yours.

**Comment by adi-mohan** [comment]
Hi. Could you please restart the Travis CI, as I think most of the builds got cancelled due to timeout issues.

**Comment by rhysmeister** [comment]
The roles failed in the idempotence stage. You're probably better off running the molecule test locally to sort out any problems. Please keep chat on the issue  #219 It's easier to keep track of stuff.

**Comment by cognifloyd** [comment]
Another thing that needs to be taken care of:
A couple of the roles have a var that uses `d.stat.exists` (where `d` is the var that the is_docker task registered).
For example https://github.com/ansible-collections/community.mongodb/blob/d4a6947d653d6640ba7a6c6fac0c3bebb0e44195/roles/mongodb_mongod/vars/main.yml#L2

`d.stat.exists` is `True` when the `is_docker.txt` file exists , and `False` when the file doesn't exist. You'll need to use either `==` or `!=` in your `'docker'` `virtualization_type` comparison depending on which `d.stat.exists` comparison you're replacing.

Also, collaboration is one of the key parts of open source software development. We work together to build better software. So, I hope that both of you, @adi-mohan and @meysam81, can collaborate. I didn't know @rhysmeister until I started working on this project, but he's been great to work with.

**Comment by cognifloyd** [comment]
I just merged #219

@adi-mohan and @meysam81 Thank you both for working on this. There are many other things that you can do in this and other open source projects. Please continue to contribute wherever you can. Thank you again!
